File: erp-warehouse/src/main/java/com/own/erp/warehouse/entity/Warehouse.java
package com.own.erp.warehouse.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * @author : chenyi
 * @Date : 2026/9/3
 * @Description : 仓库(warehouse)
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("warehouse")
public class Warehouse {

    /** 主键 */
    @TableId(type = IdType.AUTO)
    private Long id;

    /** 仓库名称 */
    private String whName;

    /** SELF自仓/FBA/OVERSEAS海外仓/VIRTUAL虚拟仓 */
    private String whType;

    /** 国家(ISO 3166) */
    private String country;

    /** 仓库地址 */
    private String address;

    /** 1启用0禁用 */
    private Integer status;

    /** 创建时间 */
    private LocalDateTime createdAt;

    /** 更新时间 */
    private LocalDateTime updatedAt;

    /** 逻辑删除:0=正常,非0=已删(值=被删行id);delval=id 配合唯一键含 deleted,删后同键可重建(TODO#7) */
    @TableLogic(value = "0", delval = "id")
    private Long deleted;
}

File: erp-web/src/api/interface/warehouse/warehouse.ts
/**
 * 本文件由 pnpm gen:page 生成(spec 行式拍板 + tools/openapi.json 快照)
 * 默认存在即跳过:人工改动不会被 --force 之外的任何方式覆盖;重新生成前先 diff 人工改动
 * 框架代码禁手改;业务槽位一律 TODO(编号),编号已登记 TODO.md
 */

/** 仓库管理接口类型(gen:page 从 openapi 快照生成;对齐后端契约,禁手抄字段) */

/** 仓库管理实体(分页行 / 详情) */
export interface WarehouseResponse {
  /** id */
  id: number
  /** 仓库名称 */
  whName: string
  /** 仓库类型 */
  whType: string
  /** 国家 */
  country: string
  /** 仓库地址 */
  address: string
  /** 状态 */
  status: number
  /** 创建时间 */
  createdAt: string
  /** updatedAt */
  updatedAt: string
}

/** 仓库管理新增/修改入参 */
export interface WarehouseSaveRequest {
  /** whName */
  whName?: string
  /** whType */
  whType?: string
  /** country */
  country?: string
  /** address */
  address?: string
  /** status */
  status?: number
}
